config/orte_check_lsf.m4

```diff
@@ -120,6 +120,70 @@ AC_DEFUN([ORTE_CHECK_LSF],[
                         [orte_check_lsf_happy="yes"],
                         [orte_check_lsf_happy="no"])])
 
+
+          # Some versions of LSF ship with a libevent.so in their library path.
+          # This is _not_ a copy of Libevent, but something specific to their project.
+          # The Open MPI components should not need to link against LSF's libevent.so
+          # However, the presence of it in the linker search path can cause a problem
+          # if there is a system installed Libevent and Open MPI chooses the 'external'
+          # event component prior to this stage.
+          #
+          # Add a check here to see if we are in a scenario where the two are conflicting.
+          # In which case the earlier checks for successful compile of an LSF program will
+          # have failed with messages like:
+          #   lib64/libevent_pthreads.so: undefined reference to `evthread_set_condition_callbacks'
+          #   lib64/libevent_pthreads.so: undefined reference to `event_mm_malloc_'
+          #   lib64/libevent_pthreads.so: undefined reference to `event_mm_free_'
+          #   lib64/libevent_pthreads.so: undefined reference to `evthread_set_id_callback'
+          #   lib64/libevent_pthreads.so: undefined reference to `evthread_set_lock_callbacks'
+          # Because it picked up -levent from LSF, but -levent_pthreads from Libevent.
+          #
+          # So look for a function that libevent_pthreads is looking for from libevent.so
+          # If it does not appear here when adding the same option then we have
+          # a conflict.
+          orte_check_lsf_event_conflict=na
+          AS_IF([test "$orte_check_lsf_happy" = "no"],
+                [AS_IF([test "$opal_event_external_support" = "yes"],
+                       [AS_IF([test "$orte_check_lsf_libdir" = "" ],
+                              [],
+                              [LDFLAGS="$LDFLAGS -L$orte_check_lsf_libdir"])
+                        # Note that we do not want to set LIBS here to include -llsf since
+                        # the check is not for an LSF library, but for the conflict with
+                        # LDFLAGS.
+                        AC_CHECK_LIB([event], [evthread_set_condition_callbacks],
+                                     [AC_MSG_CHECKING([for libevent conflict])
+                                      AC_MSG_RESULT([No. The correct libevent.so was linked.])
+                                      orte_check_lsf_event_conflict=no],
+                                     [AC_MSG_CHECKING([for libevent conflict])
```

Comment:
(Note from code review with Geoff)
We use `event` here instead of `event_core` because we are looking to see if the `-levent` that was picked up was from LSF (false condition) or Libevent (true condition) to detect if there is a conflict.

If we use `-levent_core` instead of `-levent` then this condition will fire if we built with an external libevent on a system without LSF installed. Thus triggering a warning that is inaccurate. So if we decide to go with `-levent_core` we should probably remove this check since it will be for an always false case.

Note that if we use `-levent` instead on a system with an external libevent without LSF installed then we get the 'true' condition and no warning is issued.

Reply:
Actually I take that back. I got the logic backwards for the `libevent_core` scenario.
 - On a system with no LSF and an external libevent, the 'true' path is followed and no warning is reported. That's because it'll link in the libevent.so from Libevent.
 - On a system with LSF in the default path and an external libevent, the check would only fire if LSF couldn't compile. In that case it could run the AC_CHECK_LIB and the 'false' path is followed (since the only libevent.so it could pick up is from LSF).

In the second case the warning message is misleading. I think I can protect this a bit better. I'll tinker with it. At the very least I need to add a clarifying comment.

Reply:
I just pushed a change to add protection for the `libevent_core` case. The check now only runs if it detects a `-levent` in the `$LIBS` - thus exposing us to picking up the wrong library.